docs/develop/troubleshoot-sso-in-office-add-ins.md

```diff
@@ -21,128 +21,123 @@ We strongly recommend that you use a tool that can intercept and display the HTT
+Invalid Grant. This usually means that Office has not been pre-authorized to the add-in's web service. For more information, see [Create the service application](sso-in-office-add-ins.md#create-the-service-application) and [Register the add-in with Azure AD v2.0 endpoint](register-sso-add-in-aad-v2.md). This also may happen if the user has not granted your service application permissions to their `profile`, or has revoked consent. Your code should fall back to an alternate system of user authentication.
+
+Another possible cause, during development, is that your add-in using Internet Explorer, and you are using a self-signed certificate. (To determine which browser is being used by the add-in, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).)
 
 ### 13006
 
-Client Error. Your code should suggest that the user sign out and restart Office, or restart the Office browser session.
+Client Error. This error is only seen in **Office on the web**. Your code should suggest that the user sign out and then restart the Office browser session.
 
 ### 13007
 
 The Office host was unable to get an access token to the add-in's web service.
 
-- If this error occurs during development, be sure that your add-in registration and add-in manifest specify the `openid` and `profile` permissions. For more information, see [Register the add-in with Azure AD v2.0 endpoint](create-sso-office-add-ins-aspnet.md#register-the-add-in-with-azure-ad-v20-endpoint) (ASP.NET) or [Register the add-in with Azure AD v2.0 endpoint](create-sso-office-add-ins-nodejs.md#register-the-add-in-with-azure-ad-v20-endpoint) (Node JS), and [Configure the add-in](create-sso-office-add-ins-aspnet.md#configure-the-add-in) (ASP.NET) or [Configure the add-in](create-sso-office-add-ins-nodejs.md#configure-the-add-in) (Node JS).
+- If this error occurs during development, be sure that your add-in registration and add-in manifest specify the `profile` permission (and the `openid` permission, if you are using MSAL.NET). For more information, see [Register the add-in with Azure AD v2.0 endpoint](register-sso-add-in-aad-v2.md).
 - In production, there are several things that can cause this error. Some of them are:
-    - The user has revoked consent, after previously granting it. Your code should recall the `getAccessTokenAsync` method with the option `forceConsent: true`, but no more than once.
-    - The user is has an Microsoft Account (MSA) identity. Some situations that would cause one of the other 13nnn errors with a Work or School account, will cause a 13007 when a MSA is used.
+    - The user is has an Microsoft Account (MSA) identity.
+    - Some situations that would cause one of the other 13nnn errors with a Work or School account, will cause a 13007 when a MSA is used.
```

Comment:
I couldn't find a style guide page, but I'm used to seeing numbers replaced by X's (13xxx). Is this 'n' notation standard is web APIs?